automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala

```diff
@@ -12,24 +10,44 @@ import org.broadinstitute.dsde.workbench.fixture.BillingFixtures
 import org.broadinstitute.dsde.workbench.model._
 import org.broadinstitute.dsde.workbench.model.google.{GoogleProject, ServiceAccountName}
 import org.broadinstitute.dsde.workbench.service.SamModel._
+import org.broadinstitute.dsde.workbench.service._
 import org.broadinstitute.dsde.workbench.service.test.CleanUp
-import org.broadinstitute.dsde.workbench.service.{Orchestration, Sam, Thurloe, _}
+import org.broadinstitute.dsde.workbench.service.util.Tags
 import org.broadinstitute.dsde.workbench.test.SamConfig
 import org.scalatest.concurrent.{Eventually, ScalaFutures}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.tagobjects.Retryable
 import org.scalatest.time.{Seconds, Span}
-import org.broadinstitute.dsde.workbench.service.util.Tags
+import org.scalatest.{Canceled, Failed, Outcome, Retries}
 
+import java.util.UUID
 import scala.concurrent.Await
 import scala.concurrent.duration.{Duration, _}
-import org.scalatest.freespec.AnyFreeSpec
-import org.scalatest.matchers.should.Matchers
 
-class SamApiSpec extends AnyFreeSpec with BillingFixtures with Matchers with ScalaFutures with CleanUp with Eventually with TestKitBase {
+class SamApiSpec extends AnyFreeSpec with BillingFixtures with Matchers with ScalaFutures with CleanUp with Eventually with TestKitBase with Retries {
   implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)))
   implicit lazy val system = ActorSystem()
 
   val gcsConfig = SamConfig.GCS
 
+  val retries = 4
+
+  // We are intentionally not using trait `FailedTestRetryable` from WorkbenchLibs here because that trait only retries
+  // failures (not cancels) and if it succeeds on a retry, it will only put the test into a Canceled state instead of a
+  // passing state, which for our purposes here, is what we want.
+  override def withFixture(test: NoArgTest): Outcome = {
+    if (isRetryable(test)) withFixture(test, retries) else super.withFixture(test)
+  }
```

Comment:
These mixed function calls are a bit confusing - hard to see who is calling whom. Perhaps this might be a simplification?

```suggestion
  override def withFixture(test: NoArgTest): Outcome = {
    val maxAttempts = if (isRetryable(test)) 4 else 1

    @tailrec
    def go(attempt: Int): Outcome =
      super.withFixture(test) match {
        case failure@Exceptional(_) => if (attempt < maxAttempts) go(attempt + 1) else failure
        case other => other
      }

    go(attempt = 0)
  }
```